website/content/en/docs/building-operators/golang/advanced-topics.md

```diff
@@ -113,10 +113,192 @@ func init() {
 * After adding new import paths to your operator project, run `go mod vendor` if a `vendor/` directory is present in the root of your project directory to fulfill these dependencies.
 * Your 3rd party resource needs to be added before add the controller in `"Setup all Controllers"`.
 
-### Metrics
+### Monitoring and Observability
+This section aims to explain how to expose the first operator's custom metrics and alerts. It focuses on the technical aspects of adding metrics and alerts, and demonstrates it over the sample [memcached-operator]. For more information regarding operators' monitoring, visit [observability-best-practices].  
 
-To learn about how metrics work in the Operator SDK read the [metrics section][metrics_doc] of the Kubebuilder documentation.
+#### Prerequisites
+- Deploy [kube-prometheus] on your cluster.
+- Make sure Prometheus has access to the operator's namespace, by setting the corresponding RBAC rules. You may find [prometheus_role.yaml] and [prometheus_role_binding.yaml] examples helpful. 
+- Create a new `monitoring` directory, which will contain the metrics and alerts implementation.
```

Comment:
Please explain why.
For example: We recommend to create a new `monitoring` directory, which will contain the metrics and alerts implementation, in order to keep the monitoring code and logic separated from the core operator code. This will allow easier maintenance of both the core operator code and the monitoring code.

Reply:
updated